logging/DetailedErrorFormatter.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace SilverStripe\Framework\Logging;
+
+use Monolog\Logger;
+use Monolog\Handler\AbstractProcessingHandler;
+use Monolog\Formatter\FormatterInterface;
+
+/**
+ * Monolog-compatible error handler that will output a detailed error message to the screen.
+ */
+class DetailedErrorFormatter implements FormatterInterface
+{
+	public function format(array $record)
+	{
+		if(isset($record['context']['exception'])) {
+			$exception = $record['context']['exception'];
+			$context = array(
+				'code' => $exception->getCode(),
+				'message' => 'Uncaught ' . get_class($exception) . ': ' . $exception->getMessage(),
+				'file' => $exception->getFile(),
+				'line' => $exception->getLine(),
+				'trace' => $exception->getTrace(),
+			);
+		} else {
+			$context = $record['context'];
+			foreach(array('code','message','file','line') as $key) {
+				if(!isset($context[$key])) {
+					$context[$key] = isset($record[$key]) ? $record[$key] : null;
+				}
+			}
+
+			$trace = debug_backtrace();
```

Comment:
For fatal errors this is pointless, since it's registered as a `register_shutdown_function`. This gives the following stacktrace.

```
SilverStripe\Framework\Logging\DetailedErrorFormatter->format(Array) 
AbstractProcessingHandler.php:35
Monolog\Handler\AbstractProcessingHandler->handle(Array) 
Logger.php:289
Monolog\Logger->addRecord(550,Fatal Error (E_USER_ERROR): OH NO,Array) 
Logger.php:467
Monolog\Logger->log(alert,Fatal Error (E_USER_ERROR): OH NO,Array) 
ErrorHandler.php:176
Monolog\ErrorHandler->handleFatalError() 
```